go/libraries/doltcore/sqle/database_provider.go

```diff
@@ -233,7 +240,11 @@ func (p DoltDatabaseProvider) databaseForRevision(ctx context.Context, revDB str
 	}
 
 	if doltdb.IsValidCommitHash(revSpec) {
-		db, init, err := dbRevisionForCommit(ctx, srcDb, revSpec)
+		srcDb, ok = srcDb.(Database)
+		if !ok {
+			return nil, dsess.InitialDbState{}, false, ErrCannotCreateReplicaRevisionDbForCommit
```

Comment:
No reason to error here that I can see. The replica DB may have the commit hash named, or it may not. Shouldn't err out here by default.

Add a TODO to do better error management.

Reply:
kk. i used the `USE db/branch` syntax to test, wasn't sure whether or not that's wrapped in a transaction

Reply:
it was just returning nil before, l'll set it back to that